components/match2/commons/match_group_util.lua

```diff
@@ -438,92 +481,36 @@ function MatchGroupUtil.gameFromRecord(record)
 	}
 end
 
-function MatchGroupUtil.computeUpperMatchIds(matchesById)
-	local upperMatchIds = {}
-	for matchId, match in pairs(matchesById) do
-		if match.bracketData.type == 'bracket' then
-			for _, lowerMatchId in ipairs(match.bracketData.lowerMatchIds) do
-				upperMatchIds[lowerMatchId] = matchId
-			end
-		end
+function MatchGroupUtil.computeLowerMatchIdsFromLegacy(data)
+	local lowerMatchIds = {}
+	if nilIfEmpty(data.toupper) then
+		table.insert(lowerMatchIds, data.toupper)
 	end
-
-	-- Matches without upper matches
-	local rootMatchIds = {}
-	for matchId, _ in pairs(matchesById) do
-		if not upperMatchIds[matchId]
-			and not StringUtils.endsWith(matchId, 'RxMBR') then
-			table.insert(rootMatchIds, matchId)
-		end
+	if nilIfEmpty(data.tolower) then
+		table.insert(lowerMatchIds, data.tolower)
 	end
-
-	-- Use custom ordering specified by rootIndex if present
-	Array.sortInPlaceBy(rootMatchIds, function(matchId)
-		return {matchesById[matchId].bracketData.rootIndex or -1, matchId}
-	end)
-
-	return upperMatchIds, rootMatchIds
+	return lowerMatchIds
 end
 
-function MatchGroupUtil.dfsFrom(bracketDatasById, start)
-	return TreeUtil.dfs(
-		function(matchId)
-			return bracketDatasById[matchId].lowerMatchIds
-		end,
-		start
-	)
-end
-
-function MatchGroupUtil.computeDepthsFrom(bracketDatasById, startMatchId)
-	local depths = {}
-	local maxDepth = -1
-	local function visit(matchId, depth)
-		local bracketData = bracketDatasById[matchId]
-		depths[matchId] = depth
-		maxDepth = math.max(maxDepth, depth + bracketData.skipRound)
-		for _, lowerMatchId in ipairs(bracketData.lowerMatchIds) do
-			visit(lowerMatchId, depth + 1 + bracketData.skipRound)
-		end
-	end
-	visit(startMatchId, 0)
-	return depths, maxDepth + 1
-end
-
--- TODO store and read this from LPDB
-function MatchGroupUtil.computeRounds(bracketDatasById, rootMatchIds)
-	local rounds = {}
-	local roundPropsByMatchId = {}
-	for _, rootMatchId in ipairs(rootMatchIds) do
-		local depths, depthCount = MatchGroupUtil.computeDepthsFrom(bracketDatasById, rootMatchId)
-		for _ = #rounds + 1, depthCount do
-			table.insert(rounds, {})
-		end
-
-		for matchId, depth in pairs(depths) do
-			roundPropsByMatchId[matchId] = {
-				depth = depth,
-				depthCount = depthCount,
-			}
+function MatchGroupUtil.autoAssignLowerEdges(lowerMatchCount, opponentCount)
+	local lowerEdges = {}
+	if lowerMatchCount <= opponentCount then
```

Comment:
This condition could use some explanation, it's hard to understand based off of the var names.